beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs

```diff
@@ -330,13 +330,15 @@ impl Eq for Score {}
 
 impl PartialOrd for Score {
     fn partial_cmp(&self, other: &Score) -> Option<std::cmp::Ordering> {
-        self.score().partial_cmp(&other.score())
+        Some(self.cmp(other))
     }
 }
 
 impl Ord for Score {
     fn cmp(&self, other: &Score) -> std::cmp::Ordering {
-        self.partial_cmp(other).unwrap_or(std::cmp::Ordering::Equal)
+        self.score()
+            .partial_cmp(&other.score())
+            .unwrap_or(std::cmp::Ordering::Equal)
```

Comment:
Clippy thought the old impl was pretty [_suspicious_](https://rust-lang.github.io/rust-clippy/master/index.html#/incorrect_partial_ord_impl_on_ord_type).

The effect of Clippy's suggested change is that `partial_ord` will never return `None` for incomparable values (like `NaN`). I think this is OK, as we shouldn't be encountering this anyway (fingers crossed).

Reply:
It doesn't look like there are any other scenarios (other than comparing `NaN`)  where `f64::partial_cmp` would return None, so this is probably fine 🤞